src/main/java/org/example/daos/MySqIJobRoleDao.java

```diff
@@ -60,4 +62,38 @@ public List<JobRole> getJobRoles() throws JobRoleDaoException {
         }
         return jobRoles;
     }
+
+    @Override
+    public SingleJobRoleResponse getJobRoleById(final int id)
+            throws JobRoleDaoException {
+        try (Connection connection = databaseConnector.getConnection()) {
+            String query = "Select role_name, location, capability,"
+            + " band, closing_date, role_status, description,"
+            + " responsibilities, job_link FROM roles WHERE role_id = ?;";
+            PreparedStatement statement = connection.prepareStatement(query);
+
+            statement.setInt(1, id);
+
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                return new SingleJobRoleResponse(
+                        resultSet.getString("role_name"),
+                        resultSet.getString("location"),
+                        resultSet.getString("capability"),
+                        resultSet.getString("band"),
+                        resultSet.getDate("closing_date"),
+                        resultSet.getInt("role_status"),
+                        resultSet.getString("description"),
+                        resultSet.getString("responsibilities"),
+                        resultSet.getString("job_link")
+                );
+
+            }
+            return null;
```

Comment:
this doesn't sound good, I don't like returning nulls. Empty object sounds better. When I see this method, it tells me it's going to return SingleJobRoleResponse. Getting a null is going to give me a debugging headache

Reply:
fixed!